x-pack/platform/plugins/shared/onechat/server/services/tools/builtin/builtin_registry.ts

```diff
@@ -6,25 +6,25 @@
  */
 
 import { validateToolId } from '@kbn/onechat-common';
-import type { BuiltinToolDefinition } from '@kbn/onechat-server';
+import type { StaticToolRegistration } from '@kbn/onechat-server/tools';
 
 export interface BuiltinToolRegistry {
-  register(tool: BuiltinToolDefinition<any>): void;
+  register(tool: StaticToolRegistration<any>): void;
```

Comment:
The built-in registry now accepts any type of static registration.

Note: except for ID checks, we are not validating the static tools - both because it's difficult so early in the init cycle, but also because we should be good assuming our internal tools are, well, valid.